test/navigation/NavigatorSpec.scala

```diff
@@ -1904,6 +1904,160 @@ class NavigatorSpec extends SpecBase {
           }
 
         }
+        "in Supplementary Unit Update Journey" - {
+
+          "must go from HasSupplementaryUnitUpdatePage" - {
+
+            "to SupplementaryUnitUpdatePage when answer is Yes and unit is not defined" in {
+
+              val answers =
+                UserAnswers(userAnswersId).set(HasSupplementaryUnitUpdatePage(testRecordId), true).success.value
+              navigator.nextPage(
+                HasSupplementaryUnitUpdatePage(testRecordId),
+                CheckMode,
+                answers
+              ) mustBe routes.SupplementaryUnitController
+                .onPageLoadUpdate(
+                  CheckMode,
+                  testRecordId
+                )
+            }
+
+            "to CyaSupplementaryUnitController when answer is Yes and unit is already defined" in {
+
+              val answers = UserAnswers(userAnswersId)
+                .set(HasSupplementaryUnitUpdatePage(testRecordId), true)
+                .success
+                .value
+                .set(SupplementaryUnitUpdatePage(testRecordId), "974.0")
+                .success
+                .value
+
+              navigator.nextPage(
+                HasSupplementaryUnitUpdatePage(testRecordId),
+                CheckMode,
```

Comment:
Should be testing NormalMode here, and the behaviour should be to show the SupplementaryUnitUpdatePage?

Reply:
fixed